src/components/buttons/ButtonIcon.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { ComponentPropsWithoutRef, forwardRef, MouseEvent, useMemo } from 'react';
+import { IconProps } from '../icons';
+import { SpinnerOverrideColor } from '../spinner/SpinnerOverrideColor';
+import { Tooltip } from '../tooltip';
+import { useButtonClickHandler } from './Button';
+import { BUTTON_VARIETY_STYLES, BUTTONICON_DIMENSIONS, ButtonIconStyled } from './ButtonStyles';
+import { ButtonSize, ButtonVariety, HTMLButtonProps } from './ButtonTypes';
+
+type TooltipProps = ComponentPropsWithoutRef<typeof Tooltip>;
+type TooltipOptions = Omit<TooltipProps, 'children' | 'content' | 'key'>;
```

Comment:
what is 'key' ?

Reply:
it's the ReactKey, added by `ComponentPropsWithoutRef`. I don't think it makes sense to expose that